sklearn/compose/_column_transformer.py

```diff
@@ -336,8 +336,12 @@ def get_feature_names(self):
                 raise AttributeError("Transformer %s (type %s) does not "
                                      "provide get_feature_names."
                                      % (str(name), type(trans).__name__))
+            try:
```

Comment:
this is ducktyping to support both transformative and non-transformative get_feature_names.